src/Unused/Util.hs

```diff
@@ -0,0 +1,12 @@
+module Unused.Util
+    ( groupBy
+    ) where
+
+import Data.List (nub)
+
+groupBy :: Eq b => (a -> b) -> [a] -> [(b, [a])]
+groupBy f l =
```

Comment:
@pbrisbin this structure feeds directly into `Data.Map.Strict.fromList`; if you think there's confusion caused by the overlap, I'm fine changing the function name